test/wsfed.tests.js

```diff
@@ -171,6 +204,100 @@ describe('wsfed', function () {
   });
 
   describe('using custom profile mapper', function() {
+    describe('when NameIdentifier and NameIdentifierFormat have been configured', function() {
+      const fakeNameIdentifier = 'fakeNameIdentifier';
+      const fakeNameIdentifierFormat = 'fakeNameIdentifierFormat';
+
+      function ProfileMapper(user) {
+        this.user = user;
+      }
+      ProfileMapper.prototype.getClaims = function () {
+        return this.user;
+      }
+      ProfileMapper.prototype.getNameIdentifier = function () {
+        return {
+          nameIdentifier: fakeNameIdentifier,
+          nameIdentifierFormat: fakeNameIdentifierFormat,
+        };
+      }
+
+      before(function () {
+        server.options = {
+          profileMapper: function createProfileMapper(user) {
+            return new ProfileMapper(user);
+          }
+        };
+      });
+
+      describe('when nameIdentifierFormat option has been passed', function() {
+
+        const fakeOptionNameIdentifierFormat = 'urn:oasis:names:tc:SAML:1.1:nameid-format:swfedfakeformat';
+        var body, $, signedAssertion, attributes;
```

Comment:
If we are using `const` from ES2015, maybe we can also replace `var` by `let` in this PR at least for new tests?

Reply:
Sounds like a good idea, I've gone a little further and replaced the usage of `var` across the whole test suite let me know what you think.